front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -85,10 +85,29 @@ const Network = ({
   const [height, setHeight] = useState(0);
   const [width, setWidth] = useState(0);
   const [zoomLevel, setZoomLevel] = useState(0);
+  const [hoverNode, setHoverNode] = useState<Node | undefined>();
+  const [pointerPosition, setPointerPosition] = useState([0, 0]);
+  const [hiddenNodes, setHiddenNodes] = useState<Array<Node>>([]);
 
   const networkRef = useRef<ForceGraphMethods>();
   const { loading, network } = useNetwork(viewId);
   const view = useInsightsView(viewId);
+  const tooltipRef = document.getElementsByClassName('graph-tooltip')[0];
+  const canvasRef = useRef<HTMLCanvasElement | undefined>();
+
+  useEffect(() => {
```

Comment:
Why is this needed?

Reply:
the tooltip is the one that follows the mouse when a node is being hovered, and we are changing it to the "hide from map" message when the close icon is being hovered.
the canvas reference is being used to attach a pointer move event to track mouse position.